urllib3/_collections.py

```diff
@@ -148,11 +149,18 @@ def __init__(self, headers=None, **kwargs):
             self.extend(kwargs)
 
     def __setitem__(self, key, val):
-        return _dict_setitem(self, key.lower(), (key, val))
+        if key in self:
+            del self[key]
+        self.add(key, val)
+        return val
```

Comment:
My suspicion is that this reversion also reverts part of the purpose of the original change.